### PR TITLE
HIVE-26955: Select query fails when decimal column data type is change…

### PR DESCRIPTION
**Pull request by SourabhBadhya**
…d to string/char/varchar in Parquet

### What changes were proposed in this pull request?

Select query fails when decimal column data type is changed to string/char/varchar in Parquet

### Why are the changes needed?

To fix the failure induced by Select query when decimal column data type is changed to string/char/varchar in Parquet.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

QTest

### CODE REVIEW

**Review by kasakrisz — CHANGES_REQUESTED** [review]

File: ql/src/java/org/apache/hadoop/hive/ql/io/parquet/convert/ETypeConverter.java [hunk 1]
```diff
@@ -676,6 +676,39 @@ private int getScale(PrimitiveType type) {
...
             }
           };
+        case serdeConstants.VARCHAR_TYPE_NAME:
+          return new BinaryConverter<HiveVarcharWritable>(type, parent, index, hiveTypeInfo) {
+            @Override
+            protected HiveVarcharWritable convert(Binary binary) {
+              DecimalLogicalTypeAnnotation logicalType = (DecimalLogicalTypeAnnotation) type.getLogicalTypeAnnotation();
+              HiveDecimalWritable decimalWritable = HiveDecimalUtils.enforcePrecisionScale(
+                      new HiveDecimalWritable(binary.getBytes(), logicalType.getScale()),
+                      new DecimalTypeInfo(logicalType.getPrecision(), logicalType.getScale()));
```
[the remark this answers is not in the document]
Comment (SourabhBadhya) [reply ql/src/java/org/apache/hadoop/hive/ql/io/parquet/convert/ETypeConverter.java]: Nice way to refactor @kasakrisz . Implemented it in a similar way. Done.

**Review by kasakrisz — APPROVED** [review]
+1, pending tests

File: ql/src/test/queries/clientpositive/parquet_alter_column_type.q [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+alter table test_parquet_7 change id id char(8);
+select * from test_parquet_7;
+
+drop table test_parquet_1;
+drop table test_parquet_2;
+drop table test_parquet_3;
+drop table test_parquet_4;
+drop table test_parquet_5;
+drop table test_parquet_6;
+drop table test_parquet_7;
```
Comment (soumyakanti3578) [inline ql/src/test/queries/clientpositive/parquet_alter_column_type.q:122]: Minor nit - Ideally there should be a new line at the end of the file.

Comment (SourabhBadhya) [reply ql/src/test/queries/clientpositive/parquet_alter_column_type.q]: Done.

**Pull request merged**